### PR TITLE
=str #19025 check that the server has started before trying to connect

### LINKED ISSUE #19025: FAILED: TcpSpec
(this was before merging the Fused tcp).

https://jenkins.akka.io:8498/job/pr-validator-per-commit-jenkins/4329/consoleFull

```
[info] - must shut down properly even if some accepted connection Flows have not been subscribed to *** FAILED *** (3 seconds, 31 milliseconds)
[WARN] [11/26/2015 16:11:31.439] [TcpSpec-akka.actor.default-dispatcher-15] [akka://TcpSpec/system/IO-TCP/selectors/$a/67] received dead letter from Actor[akka://TcpSpec/deadLetters]: ChannelReadable
[info]   java.util.concurrent.TimeoutException: Futures timed out after [3 seconds]
[info]   at scala.concurrent.impl.Promise$DefaultPromise.ready(Promise.scala:219)
[info]   at scala.concurrent.impl.Promise$DefaultPromise.result(Promise.scala:223)
[info]   at scala.concurrent.Await$$anonfun$result$1.apply(package.scala:107)
[info]   at scala.concurrent.BlockContext$DefaultBlockContext$.blockOn(BlockContext.scala:53)
[info]   at scala.concurrent.Await$.result(package.scala:107)
[info]   at akka.stream.io.TcpSpec$$anonfun$2$$anonfun$apply$mcV$sp$35$$anonfun$apply$mcV$sp$36.apply$mcV$sp(TcpSpec.scala:531)
[info]   at akka.stream.io.TcpSpec$$anonfun$2$$anonfun$apply$mcV$sp$35$$anonfun$apply$mcV$sp$36.apply(TcpSpec.scala:510)
[info]   at akka.stream.io.TcpSpec$$anonfun$2$$anonfun$apply$mcV$sp$35$$anonfun$apply$mcV$sp$36.apply(TcpSpec.scala:510)
[info]   at akka.stream.testkit.Utils$.assertAllStagesStopped(Utils.scala:26)
[info]   at akka.stream.io.TcpSpec$$anonfun$2$$anonfun$apply$mcV$sp$35.apply$mcV$sp(TcpSpec.scala:510)
[info]   at akka.stream.io.TcpSpec$$anonfun$2$$anonfun$apply$mcV$sp$35.apply(TcpSpec.scala:510)
[info]   at akka.stream.io.TcpSpec$$anonfun$2$$anonfun$apply$mcV$sp$35.apply(TcpSpec.scala:510)
[info]   at org.scalatest.Transformer$$anonfun$apply$1.apply$mcV$sp(Transformer.scala:22)
[info]   at org.scalatest.OutcomeOf$class.outcomeOf(OutcomeOf.scala:85)
[info]   at org.scalatest.OutcomeOf$.outcomeOf(OutcomeOf.scala:104)
[info]   at org.scalatest.Transformer.apply(Transformer.scala:22)
[info]   at org.scalatest.Transformer.apply(Transformer.scala:20)
[info]   at org.scalatest.WordSpecLike$$anon$1.apply(WordSpecLike.scala:953)
[info]   at org.scalatest.Suite$class.withFixture(Suite.scala:1122)
[info]   at akka.stream.testkit.AkkaSpec.withFixture(AkkaSpec.scala:55)
[info]   at org.scalatest.WordSpecLike$class.invokeWithFixture$1(WordSpecLike.scala:950)
[info]   at org.scalatest.WordSpecLike$$anonfun$runTest$1.apply(WordSpecLike.scala:962)
[info]   at org.scalatest.WordSpecLike$$anonfun$runTest$1.apply(WordSpecLike.scala:962)
[info]   at org.scalatest.SuperEngine.runTestImpl(Engine.scala:306)
[info]   at org.scalatest.WordSpecLike$class.runTest(WordSpecLike.scala:962)
[info]   at akka.stream.testkit.AkkaSpec.runTest(AkkaSpec.scala:55)
[info]   at org.scalatest.WordSpecLike$$anonfun$runTests$1.apply(WordSpecLike.scala:1021)
[info]   at org.scalatest.WordSpecLike$$anonfun$runTests$1.apply(WordSpecLike.scala:1021)
[info]   at org.scalatest.SuperEngine$$anonfun$traverseSubNodes$1$1.apply(Engine.scala:413)
[info]   at org.scalatest.SuperEngine$$anonfun$traverseSubNodes$1$1.apply(Engine.scala:401)
[info]   at scala.collection.immutable.List.foreach(List.scala:318)
[info]   at org.scalatest.SuperEngine.traverseSubNodes$1(Engine.scala:401)
[info]   at org.scalatest.SuperEngine.org$scalatest$SuperEngine$$runTestsInBranch(Engine.scala:390)
[info]   at org.scalatest.SuperEngine$$anonfun$traverseSubNodes$1$1.apply(Engine.scala:427)
[info]   at org.scalatest.SuperEngine$$anonfun$traverseSubNodes$1$1.apply(Engine.scala:401)
[info]   at scala.collection.immutable.List.fo
```

### PR DESCRIPTION
**Pull request by johanandren**
Hopefully fixes #19025

### CODE REVIEW

File: akka-stream-tests/src/test/scala/akka/stream/io/TcpSpec.scala [hunk 1]
```diff
@@ -22,7 +22,6 @@ import akka.testkit.EventFilter
 import akka.testkit.AkkaSpec
 
 class TcpSpec extends AkkaSpec("akka.stream.materializer.subscription-timeout.timeout = 2s") with TcpHelper {
-  var demand = 0L
```
Comment (johanandren) [inline akka-stream-tests/src/test/scala/akka/stream/io/TcpSpec.scala]: boyscout: not used anywhere

File: akka-stream-tests/src/test/scala/akka/stream/io/TcpSpec.scala [hunk 2]
```diff
@@ -542,6 +541,8 @@ class TcpSpec extends AkkaSpec("akka.stream.materializer.subscription-timeout.ti
         val bindingFuture = Tcp().bindAndHandle(Flow[ByteString], address.getHostName, address.getPort)(mat2)
 
         // Ensure server is running
+        Await.ready(bindingFuture, 3.seconds)
```
Comment (johanandren) [inline akka-stream-tests/src/test/scala/akka/stream/io/TcpSpec.scala]: I think this raced before, trying to connect before it was actually started

Comment (drewhk) [inline akka-stream-tests/src/test/scala/akka/stream/io/TcpSpec.scala]: Yes, this was definitely a race.

**Comment by drewhk** [comment]
LGTM

**Comment by patriknw** [comment]
LGTM

**Pull request merged**